simtools/simtel/simtel_histograms.py

```diff
@@ -1,26 +1,548 @@
+        distribution or the distribution used in the simulation.This is controlled by label.
+        By using label="reference", one gets the distribution function according to a pre-defined CR
+        distribution, while by using label="simulation", the spectral index of the distribution
+        function from the simulation is used.
+
+        Parameters
+        ----------
+        label: str
+            label defining which distribution function. Possible values are: "reference", or
+            "simulation".
+
+        Returns
+        -------
+        ctao_cr_spectra.spectral.PowerLaw
+            The function describing the spectral distribution.
+        """
+
+        if label == "reference":
+            particle_distribution_function = copy.copy(IRFDOC_PROTON_SPECTRUM)
+        elif label == "simulation":
+            particle_distribution_function = self._get_simulation_spectral_distribution_function()
+        else:
+            msg = f"label {label} is not valid. Please use either 'reference' or 'simulation'."
+            self._logger.error(msg)
+            raise ValueError
+        return particle_distribution_function
+
+    def _get_simulation_spectral_distribution_function(self):
+        """
+        Get the simulation particle energy distribution according to its configuration.
+
+        Returns
+        -------
+        ctao_cr_spectra.spectral.PowerLaw
+            The function describing the spectral distribution.
+        """
+        spectral_distribution = copy.copy(IRFDOC_PROTON_SPECTRUM)
+        spectral_distribution.index = self.config["spectral_index"]
+        return spectral_distribution
+
+    def estimate_observation_time(self):
+        """
+        Estimates the observation time comprised by the number of events in the simulation.
+        It uses the CTAO reference cosmic-ray spectra, the total number of particles simulated,
+        and other information from the simulation configuration self.config.
+
+        Returns
+        -------
+        float: astropy.Quantity[time]
+            Estimated observation time based on the total number of particles simulated.
+        """
+        first_estimate = IRFDOC_PROTON_SPECTRUM.compute_number_events(
+            self.view_cone[0],
+            self.view_cone[1],
+            1 * u.s,
+            self.total_area,
+            self.energy_range[0],
+            self.energy_range[1],
+        )
+        obs_time = (self.total_num_simulated_events / first_estimate) * u.s
+        return obs_time
+
+    def _estimate_trigger_rate_uncertainty(self):
+        """
+        Estimate the trigger rate uncertainty, based on the number of simulated events.
+        Poisson statistics are assumed.
+
+        Returns
+        -------
+        astropy.Quantity[1/time]
+            Uncertainty in the trigger rate estimate.
+        """
+        return np.sqrt(self.total_num_triggered_events) / self.estimate_observation_time()
```

Comment:
Good point. Since the simulated and reference spectra are different, I should take that into account. The correct estimate would be 

sqrt(N_triggered according to reference spectrum) / time, which is
sqrt(trigger rate X time) / time, which is
sqrt(trigger rate / time)

It turned out, I was overestimating the uncertainty. In any case, a more trustworthy estimate would be to estimate the trigger rate for a series of simulation results and extract the RMS from it. This is just a first glimpse on it.

Reply:
Are you sure on the over estimation? Maybe I misunderstood, running the integration test in calculate_trigger_rate_one_file.yml gives me:

```
NFO::simtel_histograms(l622)::calculate_trigger_rates::Histogram 1:
INFO::simtel_histograms(l623)::calculate_trigger_rates::Total number of simulated events: 2000 events
INFO::simtel_histograms(l627)::calculate_trigger_rates::Total number of triggered events: 1.0 events
INFO::simtel_histograms(l633)::calculate_trigger_rates::Estimated equivalent observation time corresponding to the number of events simulated: 9.402506742292256e-05 s
INFO::simtel_histograms(l639)::calculate_trigger_rates::Simulated event rate: 2.1271e+07 Hz
INFO::simtel_histograms(l645)::calculate_trigger_rates::System trigger event rate: 9.0064e+03 ± 1.0313e+02 Hz
```

So 1 triggered event (it is a test...), shouldn't that give you more than a 2% uncertainty?

Reply:
As we discussed, I derived the uncertainty estimate analytically now using propagation of errors and Poisson statistics. Please see image below for details on the step by step:
![image](https://github.com/gammasim/simtools/assets/50831535/29422330-e421-4818-823a-5f62a697029c)